app/src/main/kotlin/com/intive/picover/common/viewmodel/state/ViewModelState.kt

```diff
@@ -0,0 +1,19 @@
+package com.intive.picover.common.viewmodel.state
+
+sealed class ViewModelState<out T> {
+	object Loading : ViewModelState<Nothing>()
+	object Error : ViewModelState<Nothing>()
+	data class Loaded<T>(val data: T) : ViewModelState<T>()
+
+	fun isLoading() =
+		this is Loading
+
+	fun isError() =
+		this is Error
+
+	fun isLoaded() =
+		this is Loaded
+
+	fun data() =
+		(this as? Loaded)?.data
```

Comment:
I'm wondering if it wouldn't be better to just force not null here

Reply:
Yes, i was thinking also about this version :D adjusted this, as i see 2 votes for that solution :D